File: Lista01/lab1.ml

let x_1 (a, b, c, d) = a;;
let x_2 (a, b, c, d) = b;;
let x_3 (a, b, c, d) = c;;
let x_4 (a, b, c, d) = d;;

let mirror4 xss =
  (x_2 xss, x_1 xss, x_4 xss, x_3 xss);;

mirror4 (1, 2, 3, 4);;

let rec substitute (xss, a, b) =
  if xss = [] then []
  else if List.hd xss = a then b :: substitute (List.tl xss, a, b)
  else List.hd xss :: substitute (List.tl xss, a, b);;

substitute ([1; 2; 3; 4; 5], 3, 8);;


let rec remove (xss, pos) =
  if pos = 0 then List.tl xss
  else List.hd xss :: remove (List.tl xss, pos - 1);;

remove ([1; 2; 3; 4; 5], 3);;
remove ([1; 2; 3; 4; 5], 8);;

File: Lista01/lab1.scala
def mirror4[A](xss: (A, A, A, A)): (A, A, A, A) =
  (xss._2, xss._1, xss._4, xss._3)

mirror4((1, 2, 3, 4))
mirror4(2.5, 1, "Ala", 'a')
mirror4((2.5, 1, "Ala", 'a'))

def substitute[A](xss: List[A], a: A, b: A): List[A] =
  if xss == Nil then Nil
  else if xss.head == a then b :: substitute(xss.tail, a, b)
  else xss.head :: substitute(xss.tail, a, b)

substitute(List(1, 2, 3, 4), 3, 8)


def remove[A](xss: List[A], pos: Int): List[A] =
  if pos == 0 then xss.tail
  else xss.head :: remove(xss.tail, pos - 1)

remove(List(1, 2, 3, 4, 5), 3)